src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/TypeCast.cs

```diff
@@ -813,24 +813,43 @@ private static unsafe void StelemRef_Helper(ref object element, MethodTable* ele
             }
         }
 
+        private static unsafe ref object ThrowArrayMismatchException(Array array)
```

Comment:
Why does this return `ref object` ? The method just throws, and the name suggests that it's exactly what it should do always.

Reply:
It's a copy-paste from CoreClr's corelib - I think it allows the method to be tail called and `LdelemaRef` to avoid having a frame as a result. The frameless property likely doesn't apply to nativeAOT right now because we need a call to check type equivalence (CoreCLR doesn't, it just compares pointers). But it sets us up for easier code sharing, so I kept the structure.

Reply:
Could you add a comment why it's done that way? (so that somebody doesn't "simplify" it later on)